lighthouse-core/gather/gatherers/image-elements.js

```diff
@@ -162,6 +169,70 @@ function determineNaturalSize(url) {
   });
 }
 
+/**
+ * @param {LH.Crdp.CSS.CSSStyle} [style]
+ * @param {string} property
+ * @return {string | undefined}
+ */
+function findSizeDeclaration(style, property) {
+  if (!style) return;
+  const definedProp = style.cssProperties.find(({name}) => name === property);
+  if (!definedProp) return;
+  return definedProp.value;
+}
+
+/**
+ * Finds the most specific directly matched CSS font-size rule from the list.
```

Comment:
this function is nearly identical to the font-size one, maybe we can modify that one so that `hasFontSizeDeclaration` is a parameter to pass in (`isDeclarationOfInterest` or something) and just reference it here with 

```js
return FontSize.findMostSpecificCSSRule(matchedCSSRules, declaration => findSizeDeclaration(declaration, property)).style.cssProperties.find(({name}) => name === property).value
```

but cleaned up and not all in one line of course ;)

Reply:
sounds great! I'll try this